### PR TITLE
QR-Code detection: accuracy improvement

### PR DESCRIPTION
**Pull request by APrigarina**
**Merge with extra**: https://github.com/opencv/opencv_extra/pull/672

### CODE REVIEW

File: modules/objdetect/src/qrcode.cpp [hunk 1]
```diff
@@ -402,6 +453,7 @@ bool QRDetect::computeTransformationPoints()
         }
     }
 
+
```
Comment (allnes) [inline modules/objdetect/src/qrcode.cpp]: @APrigarina please remove unnecessary additions

File: modules/objdetect/src/qrcode.cpp [hunk 2]
```diff
@@ -1138,5 +1212,4 @@ cv::String QRCodeDetector::detectAndDecode(InputArray in,
     return decoded_info;
 }
 
-
```
Comment (allnes) [inline modules/objdetect/src/qrcode.cpp]: Again please remove unnecessary additions

File: modules/objdetect/src/qrcode.cpp [hunk 3]
```diff
@@ -557,6 +609,7 @@ vector<Point2f> QRDetect::getQuadrilateral(vector<Point2f> angle_list)
 
     vector<Point> integer_hull;
     convexHull(locations, integer_hull);
+
```
Comment (allnes) [inline modules/objdetect/src/qrcode.cpp]: Again please remove unnecessary additions

File: modules/objdetect/src/qrcode.cpp [hunk 4]
```diff
@@ -53,18 +54,28 @@ void QRDetect::init(const Mat& src, double eps_vertical_, double eps_horizontal_
     const double min_side = std::min(src.size().width, src.size().height);
     if (min_side < 512.0)
     {
+        resize_direction = -1;
         coeff_expansion = 512.0 / min_side;
         const int width  = cvRound(src.size().width  * coeff_expansion);
-        const int height = cvRound(src.size().height  * coeff_expansion);
```
Comment (allnes) [inline modules/objdetect/src/qrcode.cpp]: Again please remove unnecessary additions, better to do it everywhere

Comment (allnes) [reply modules/objdetect/src/qrcode.cpp]: please correct this line

File: modules/objdetect/src/qrcode.cpp [hunk 5]
```diff
@@ -53,14 +54,25 @@ void QRDetect::init(const Mat& src, double eps_vertical_, double eps_horizontal_
     const double min_side = std::min(src.size().width, src.size().height);
     if (min_side < 512.0)
     {
+        resize_direction = -1;
```
[the remark this answers is not in the document]
Comment (allnes) [reply modules/objdetect/src/qrcode.cpp]: pls call values for resize_direction (use enum)

**Comment by allnes** [comment]
@alalek please review.

**Comment by APrigarina** [comment]
@allnes please review

File: modules/objdetect/src/qrcode.cpp [hunk 6]
```diff
@@ -314,9 +345,9 @@ bool QRDetect::localization()
     CV_TRACE_FUNCTION();
     Point2f begin, end;
     vector<Vec3d> list_lines_x = searchHorizontalLines();
-    if( list_lines_x.empty() ) { return false; }
+    if( list_lines_x.empty() ) { return false;}
```
Comment (allnes) [inline modules/objdetect/src/qrcode.cpp]: not used changes

**Pull request merged**